GDJS/Runtime/ResourceLoader.ts

```diff
@@ -553,4 +565,79 @@ namespace gdjs {
       return this._model3DManager;
     }
   }
+
+  type PromiseError<T> = { item: T; reason: any };
+
+  type PromisePoolOutput<T, U> = {
+    results: Array<U>;
+    errors: Array<PromiseError<T>>;
+  };
+
+  const promisePool = <T, U>(
```

Comment:
My rule of thumb: a function is almost always verb + something. I suggest:
```suggestion
  const processWithPromisePool = <T, U>(
```